mlflow/insights/utils.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any
+
+from mlflow.exceptions import MlflowException
+from mlflow.insights.models import EvidenceEntry
+
+
+def normalize_evidence(
+    evidence: list[dict[str, Any]] | list[EvidenceEntry] | None, for_issue: bool = False
+) -> list[EvidenceEntry]:
+    """
+    Normalize evidence entries by converting them to EvidenceEntry objects and leverage type
+    and structural validation during the conversion to ensure data integrity in APIs.
+
+    Args:
+        evidence: List of evidence as dicts or EvidenceEntry objects
+        for_issue: If True, creates entries for issues (supports=None)
+                  If False, creates entries for hypotheses (with supports field)
+
+    Returns:
+        List of validated EvidenceEntry objects
+
+    Raises:
+        MlflowException: If evidence format is invalid
+    """
+    if not evidence:
+        return []
+
+    validated_entries = []
+
+    for entry in evidence:
+        if isinstance(entry, EvidenceEntry):
+            if for_issue:
+                entry.supports = None
+            validated_entries.append(entry)
+
+        elif isinstance(entry, dict):
+            try:
+                if for_issue:
+                    validated_entries.append(
+                        EvidenceEntry.for_issue(
+                            trace_id=entry.get("trace_id", ""), rationale=entry.get("rationale", "")
```

Comment:
Is an empty string allowed?

Reply:
Good call- this should be validated in the model itself. EvidenceEntry already validates that this can't be empty. We should give users a bit more of a clean exception here... I'll update this to be cleaner / more useful.